app/src/main/java/org/odk/odknotifications/DatabaseCommunicator/ServerDatabaseCommunicator.java

```diff
@@ -0,0 +1,261 @@
+package org.odk.odknotifications.DatabaseCommunicator;
+
+import android.content.ContentValues;
+import android.util.Log;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.odk.odknotifications.Model.Group;
+import org.odk.odknotifications.Model.Notification;
+import org.odk.odknotifications.Model.Response;
+import org.opendatakit.database.data.OrderedColumns;
+import org.opendatakit.database.data.TypedRow;
+import org.opendatakit.database.data.UserTable;
+import org.opendatakit.database.service.DbHandle;
+import org.opendatakit.database.service.UserDbInterface;
+import org.opendatakit.exception.ActionNotAuthorizedException;
+import org.opendatakit.exception.ServicesAvailabilityException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ServerDatabaseCommunicator {
```

Comment:
This also has to be singleton and also make this class as thread safe class

Reply:
Done.